tests/OmniSharp.MSBuild.Tests/ProjectWithAnalyzersTests.cs

```diff
@@ -17,30 +17,57 @@ public ProjectWithAnalyzersTests(ITestOutputHelper output)
         {
         }
 
+        [Fact]
+        public async Task WhenProjectIsRestoredThenReanalyzeProject()
+        {
+            using (var testProject = await TestAssets.Instance.GetTestProjectAsync("ProjectWithAnalyzers"))
+            using (var host = CreateMSBuildTestHost(testProject.Directory, configurationData: TestHelpers.GetConfigurationDataWithAnalyzerConfig(roslynAnalyzersEnabled: true)))
+            {
+                await host.RestoreProject(testProject);
+
+                var diagnostics = await host.RequestCodeCheckAsync(Path.Combine(testProject.Directory, "Program.cs"));
+
+                Assert.NotEmpty(diagnostics.QuickFixes);
+                Assert.Contains(diagnostics.QuickFixes, x => x.ToString().Contains("IDE0060")); // Unused args.
+            }
+        }
+
         [Fact]
         public async Task WhenProjectHasAnalyzersItDoesntLockAnalyzerDlls()
         {
             using (var testProject = await TestAssets.Instance.GetTestProjectAsync("ProjectWithAnalyzers"))
             {
+                // TODO: Restore when host is running doesn't reload new analyzer references yet, move this
+                // after host start after that is fixed.
                 await RestoreProject(testProject);
 
                 using (var host = CreateMSBuildTestHost(testProject.Directory, configurationData: TestHelpers.GetConfigurationDataWithAnalyzerConfig(roslynAnalyzersEnabled: true)))
                 {
-                    var diagnostics = await host.RequestCodeCheckAsync(Path.Combine(testProject.Directory, "Program.cs"));
                     var analyzerReferences = host.Workspace.CurrentSolution.Projects.Single().AnalyzerReferences.ToList();
 
                     Assert.NotEmpty(analyzerReferences);
-                    Assert.NotEmpty(diagnostics.QuickFixes);
 
                     // This should not throw when analyzers are shadow copied.
                     Directory.Delete(Path.Combine(testProject.Directory, "./nugets"), true);
                 }
             }
         }
 
+        [Fact]
+        public async Task LoadProjectWithNewCompilationOptions()
```

Comment:
could you rename this test to a better name please?
Imagine if it fails one day, it doesn't say much what it is - it would take a while to grasp that it's about diagnostic options from a ruleset 😀

Reply:
+1

---
In reply to: [281243133](https://github.com/OmniSharp/omnisharp-roslyn/pull/1440#discussion_r281243133) [](ancestors = 281243133)

Reply:
I think i copied `LoadProjectWithNewCompilationOptions` as test base and forgot to rename it 😄 I will update this to something more meaningfull.

Reply:
Renamed LoadProjectWithNewCompilationOptions -> WhenProjectIsLoadedThenItContainsCustomRulesetsFromCsproj.